action_plugins/play_sound/__init__.py

```diff
@@ -0,0 +1,183 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import annotations
+
+import os
+from typing import List, TYPE_CHECKING
+from xml.etree import ElementTree
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal
+
+from gremlin import event_handler, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    Value
+
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+from gremlin.error import GremlinError
+from gremlin.audio_player import AudioPlayer
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class PlaySoundFunctor(AbstractFunctor):
+
+    """Executes a Play Sound action callback."""
+
+    def __init__(self, action: PlaySoundData):
+        super().__init__(action)
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+        if not self._should_execute(value):
+            return
+
+        audio_player = AudioPlayer()
+        audio_player.play(self.data.sound_filename, int(self.data.sound_volume))
+
+
+class PlaySoundModel(ActionModel):
+
+    fileChanged = Signal()
+    soundVolumeChanged = Signal()
+
+    def __init__(
+            self,
+            data: AbstractActionData,
+            binding_model: InputItemBindingModel,
+            action_index: SequenceIndex,
+            parent_index: SequenceIndex,
+            parent: QtCore.QObject
+    ):
+        super().__init__(data, binding_model, action_index, parent_index, parent)
+
+    def _qml_path_impl(self) -> str:
+        return "file:///" + QtCore.QFile(
+            "core_plugins:play_sound/PlaySoundAction.qml"
+        ).fileName()
+
+    def _icon_string_impl(self) -> str:
+        return PlaySoundData.icon
+
+    def _get_sound_filename(self) -> str:
+        return self._data.sound_filename
+
+    def _set_sound_filename(self, value: str) -> None:
+        if str(value) == self._data.sound_filename:
+            return
+        self._data.sound_filename = str(value)
+        self.fileChanged.emit()
+
+    def _get_sound_volume(self) -> int:
+        return self._data.sound_volume
+
+    def _set_sound_volume(self, value: int) -> None:
+        if value == self._data.sound_volume:
+            return
+        self._data.sound_volume = value
+        self.soundVolumeChanged.emit()
+
+    sound_filename = Property(
```

Comment:
I would name them `soundFilename` and `soundVolume` as they are QML properties and that's their preferred variable naming convention. Essentially things I expose/use in QML follows their convention everything else is normal python.

Reply:
Got it.

Reply:
Oh I think I confused you as you seem to have flipped them around of what I meant to describe :-)

Reply:
yup :-), n/p, got it now.